src/model_config_utils.cc

```diff
@@ -2291,4 +2292,28 @@ TritonToDataType(const TRITONSERVER_DataType dtype)
   return inference::DataType::TYPE_INVALID;
 }
 
+bool
+IsOnlyInstanceGroupDiffers(
+    const inference::ModelConfig& old_config,
+    const inference::ModelConfig& new_config)
+{
+  ::google::protobuf::util::MessageDifferencer pb_diff;
+  pb_diff.IgnoreField(
+      old_config.descriptor()->FindFieldByLowercaseName("instance_group"));
+  return pb_diff.Compare(old_config, new_config);
+}
+
+bool
+IsInstanceEquivalent(
+    const inference::ModelInstanceGroup& instance_config_lhs,
+    const inference::ModelInstanceGroup& instance_config_rhs)
+{
+  ::google::protobuf::util::MessageDifferencer pb_diff;
+  pb_diff.IgnoreField(
+      instance_config_lhs.descriptor()->FindFieldByLowercaseName("name"));
+  pb_diff.IgnoreField(
+      instance_config_lhs.descriptor()->FindFieldByLowercaseName("count"));
+  return pb_diff.Compare(instance_config_lhs, instance_config_rhs);
```

Comment:
In the case of multiple instance groups, what does this function return if the order  of the groups is swapped? i.e. instance_group: [ A, B ] --> instance_group: [ B, A ]

Reply:
The type naming is a bit confusing, `const inference::ModelInstanceGroup& instance_config_lhs` is the config of an instance, not the entire instance group array.

> i.e. instance_group: [ A, B ] --> instance_group: [ B, A ]

This function will be comparing i.e. `A` vs `A` or `A` vs `B`

The order swapping is included in the test case: [shuffle the instance](https://github.com/triton-inference-server/server/pull/5635/files#diff-40ca2af021f4288f3fc6ae6071a244bca6350a0e657ab2822b61650cac266931R137-R143), which ensures no instance is created or destructed after swapping the order.